pkg/kubelet/status/status_manager.go

```diff
@@ -1158,6 +1158,8 @@ func mergePodStatus(pod *v1.Pod, oldPodStatus, newPodStatus v1.PodStatus, couldH
 
 	// ResourceClaimStatuses is not owned and not modified by kubelet.
 	newPodStatus.ResourceClaimStatuses = oldPodStatus.ResourceClaimStatuses
+	// ExtendedResourceClaimStatus is not owned and not modified by kubelet.
+	newPodStatus.ExtendedResourceClaimStatus = oldPodStatus.ExtendedResourceClaimStatus
```

Comment:
What happens if we don't set this? Will the patch requests kubelets send attempt to clear this field from <1.34 kubelets?

Do we need to do anything in `podStatusStrategy#PrepareForUpdate` to preserve an existing value in the field on pod status update so node restriction doesn't forbid updates because the old node appears to be trying to clear the status, similar to how we preserve a couple other pod fields on status update:

```go
	// the Pod QoS is immutable and populated at creation time by the kube-apiserver.
	// we need to backfill it for backward compatibility because the old kubelet dropped this field when the pod was rejected.
	if newPod.Status.QOSClass == "" {
		newPod.Status.QOSClass = oldPod.Status.QOSClass
	}

	preserveOldObservedGeneration(newPod, oldPod)
```

Reply:
Good question, I don't see ResourceClaimStatuses in podStatusStrategy#PrepareForUpdate, not sure if it is needed there.